### PR TITLE
feat(metrics): update to support Sentry 8 and new raven.js

### LINKED ISSUE #3599: Research updating to Sentry 8
We are currently using a forked version of Raven 1.x to report javascript errors with Sentry 7.

Soon we will need to update to sentry 8 and raven.js 2.

Ref https://github.com/getsentry/raven-js
Ref https://getsentry.com/welcome/ 

Need to check:

- if Raven 2.x works with Sentry 7
- Make sure our /metrics-errors endpoint supports Raven 2.x

We might be able to use Raven 2.x right away, or we might need to fork it in the same way to support custom endpoints.

### PR DESCRIPTION
**Pull request by vladikoff**
Fixes #3599

### CODE REVIEW

**Comment by vladikoff** [comment]
@shane-tomlinson r?

To test this, you can use the creds I sent you in the email with subject: [content server] Dev Sentry 8 creds

File: server/lib/configuration.js [hunk 1]
```diff
@@ -439,6 +439,11 @@ var conf = module.exports = convict({
       doc: 'Sentry API key',
       format: String
     },
+    api_secret: {
```
Comment (shane-tomlinson) [inline server/lib/configuration.js]: Does a corresponding entry need to be added to fxa-dev, or to puppet config?

File: tests/server/metrics-errors.js [hunk 2]
```diff
@@ -36,18 +36,29 @@ define([
     }, dfd.reject.bind(dfd)));
   };
 
-  suite['#get /metrics-errors - returns 200 without query'] = function () {
```
Comment (shane-tomlinson) [inline tests/server/metrics-errors.js]: Should we keep the `GET` variant around to handle clients that loaded before the train cut and are still reporting to the old endpoint?

File: server/lib/routes/post-metrics-errors.js [hunk 3]
```diff
@@ -76,21 +81,21 @@ function reportError(query) {
...
   return {
-    method: 'get',
+    method: 'post',
     path: '/metrics-errors',
     process: function (req, res) {
       // respond right away
       res.json({ success: true });
 
-      reportError(req.query);
+      reportError(req.query, req.body);
```
Comment (shane-tomlinson) [inline server/lib/routes/post-metrics-errors.js]: What data comes in the query parameters in the new scheme?

**Comment by shane-tomlinson** [comment]
The functionality works well, I like that POST is used instead of GET. Should we keep the GET endpoint around for a train or two as a migration period to until clients mostly stop sending errors to the previous endpoint? Will we need to add any config options to fxa-dev or to puppet-config?

@jrgm - any feedback here?

File: tests/server/metrics-errors.js (hunk 2 above)
Comment (vladikoff) [inline tests/server/metrics-errors.js]: @shane-tomlinson I thought about it but it is probably not worth it. The normal DNS switch takes just a few minutes. If we miss some of the reports it's not the end of the world.

File: server/lib/configuration.js (hunk 1 above)
Comment (vladikoff) [inline server/lib/configuration.js]: not fxa-dev, but puppet config. PR here : https://github.com/mozilla-services/puppet-config/pull/1984

File: tests/server/metrics-errors.js (hunk 2 above)
Comment (shane-tomlinson) [inline tests/server/metrics-errors.js]: I was thinking for the people that load fxa on the last train that uses the GET endpoint and then keep the window open, which seems to happen regularly.

Comment (vladikoff) [inline tests/server/metrics-errors.js]: Yeah, they will just get a 404 and we would ignore their report

Comment (vladikoff) [inline tests/server/metrics-errors.js]: from mtg: just make `GET` return 200.

File: server/lib/routes/post-metrics-errors.js (hunk 3 above)
Comment (vladikoff) [inline server/lib/routes/post-metrics-errors.js]: Same data as the query data before.

**Pull request merged**